server/eventlog_test.go

```diff
@@ -689,6 +692,137 @@ func TestParseSecureBootState(t *testing.T) {
+		{cel.EnvVarType, cel.CosRTMR, []byte("empty=")},
+		{cel.ArgType, cel.CosRTMR, []byte("--x")},
+		{cel.ArgType, cel.CosRTMR, []byte("--y")},
+		{cel.ArgType, cel.CosRTMR, []byte("")},
+		{cel.MemoryMonitorType, cel.CosRTMR, []byte{1}},
+	}
+
+	expectedEnvVars := make(map[string]string)
+	expectedEnvVars["foo"] = "bar"
+	expectedEnvVars["bar"] = "baz"
+	expectedEnvVars["baz"] = "foo=bar"
+	expectedEnvVars["empty"] = ""
+
+	wantContainerState := attestpb.ContainerState{
+		ImageReference: string(testCELEvents[0].eventPayload),
+		ImageDigest:    string(testCELEvents[1].eventPayload),
+		RestartPolicy:  attestpb.RestartPolicy_Always,
+		ImageId:        string(testCELEvents[3].eventPayload),
+		EnvVars:        expectedEnvVars,
+		Args:           []string{string(testCELEvents[8].eventPayload), string(testCELEvents[9].eventPayload), string(testCELEvents[10].eventPayload)},
+	}
+	enabled := true
+	wantHealthMonitoringState := attestpb.HealthMonitoringState{
+		MemoryEnabled: &enabled,
+	}
+
+	for _, testEvent := range testCELEvents {
+		cosEvent := cel.CosTlv{EventType: testEvent.cosNestedEventType, EventContent: testEvent.eventPayload}
+		if err := coscel.AppendEventRTMR(fakeRTMR, testEvent.register, cosEvent); err != nil {
+			t.Fatal(err)
+		}
+	}
+	buf = bytes.Buffer{}
+	if err := coscel.EncodeCEL(&buf); err != nil {
+		t.Fatal(err)
+	}
+
+	rtmrBank = register.RTMRBank{}
+	for i, digest := range getRTMR(t, fakeRTMR) {
+		rtmrBank.RTMRs = append(rtmrBank.RTMRs, register.RTMR{Index: i, Digest: digest})
+	}
+
+	if msState, err := ParseCosCanonicalEventLogRTMR(buf.Bytes(), rtmrBank); err != nil {
+		t.Errorf("expecting no error from parseCosEventLogRTMR(), but get %v", err)
+	} else {
+		if diff := cmp.Diff(msState.Cos.Container, &wantContainerState, protocmp.Transform()); diff != "" {
+			t.Errorf("unexpected container state difference:\n%v", diff)
+		}
+		if diff := cmp.Diff(msState.Cos.HealthMonitoring, &wantHealthMonitoringState, protocmp.Transform()); diff != "" {
+			t.Errorf("unexpected health monitoring state difference:\n%v", diff)
+		}
+	}
+
+	// Faking PCR with RTMR
+	imposterPcrBank := map[uint32][]byte{}
+	imposterPcrBank[1] = rtmrBank.RTMRs[0].Digest
+	imposterPcrBank[2] = rtmrBank.RTMRs[1].Digest
+	imposterPcrBank[3] = rtmrBank.RTMRs[2].Digest
+	imposterPcrBank[4] = rtmrBank.RTMRs[3].Digest
+	imposterPcrs := &pb.PCRs{Hash: pb.HashAlgo_SHA384, Pcrs: imposterPcrBank}
+
+	if msState, err = ParseCosCanonicalEventLogPCR(buf.Bytes(), imposterPcrs); err != nil {
+		t.Errorf("expecting no error from parseCosEventLogRTMR(), but get %v", err)
+	} else {
+		if diff := cmp.Diff(msState.Cos.Container, &attestpb.ContainerState{}, protocmp.Transform()); diff != "" {
+			t.Error("expect no claims when replaying against imposter PCR")
+		}
+		if diff := cmp.Diff(msState.Cos.HealthMonitoring, &attestpb.HealthMonitoringState{}, protocmp.Transform()); diff != "" {
+			t.Error("expect no claims when replaying against imposter PCR")
+		}
```

Comment:
this should go in a separate test that makes it clear what you're trying to do.

Reply:
Will refactor in the next PR